src/types/types.ts

```diff
@@ -0,0 +1,121 @@
+/**
+ * Represents options for uploading, including a function to handle previewing.
+ */
+export interface UploadOptions  { 
+  onPreview: (src: string) => void 
+};
+
+/** 
+ * User configuration of Image block tunes. Allows to add custom tunes through the config
+*/
+export interface ActionConfig { name: string; icon: string; title: string; toggle: boolean, action?: Function };
+
+/**
+ * UploadResponseFormat interface representing the response format expected from the backend on file uploading.
+ */
+export interface UploadResponseFormat {
+  /**
+   * success - 1 for successful uploading, 0 for failure 
+   */
+  success: number;
+  /**
+   * Object with file data.
+   *             'url' is required,
+   *             also can contain any additional data that will be saved and passed back
+   */
+  file: {
+    /**
+     * The URL of the uploaded image.
+     */
+    url: string;
+  };
+}
+
+/**
+ * ImageToolData interface representing the input and output data format for the image tool.
+ */
+export interface ImageToolData {
```

Comment:
seems like custom `actions` got from the config also can be included at the ImageToolData. We need to cover that case in types

Reply:
isn't this parameters what is requested `actions?: ActionConfig[]`? or should it be something like `[key: string]: any`?

Reply:
I'm not sure, but maybe something like 

```ts
export type ImageToolData = {
   /**
    * Caption for the image.
    */
   caption: string;

   /**
    * Flag indicating whether the image has a border.
    */
   withBorder: boolean;

   /**
    * Flag indicating whether the image has a background.
    */
   withBackground: boolean;

   /**
    * Flag indicating whether the image is stretched.
    */
   stretched: boolean;

   /**
    * Object containing the URL of the image file.
    */
   file: {
     url: string;
   };
} & {
  [key: ActionConfig['name']]: boolean;
}
```

or 

```ts
export type ImageToolData<Actions = {}> = {
  /**
   * Caption for the image.
   */
  caption: string;

  /**
   * Flag indicating whether the image has a border.
   */
  withBorder: boolean;

  /**
   * Flag indicating whether the image has a background.
   */
  withBackground: boolean;

  /**
   * Flag indicating whether the image is stretched.
   */
  stretched: boolean;

  /**
   * Object containing the URL of the image file.
   */
  file: {
    url: string;
  };
} & (Actions extends Record<string, boolean> ? Actions : {});
```

Reply:
okay sure